Yosemite/YosemiteTests/Stores/ProductReview/RetrieveProductReviewFromNoteUseCaseTests.swift

```diff
@@ -59,46 +65,125 @@ final class RetrieveProductReviewFromNoteUseCaseTests: XCTestCase {
         XCTAssertEqual(parcel.product, product)
     }
 
-    func testWhenSuccessfulThenItSavesTheProductReviewToStorage() throws {
+    func test_it_uses_the_existing_Note_in_Storage_if_it_is_available() throws {
         // Given
-        let storageManager = MockupStorageManager()
-        let storage = storageManager.viewStorage
+        let useCase = makeUseCase()
+        let note = TestData.note
+        let productReview = TestData.productReview
+        let product = TestData.product
+
+        let storageNote = viewStorage.insertNewObject(ofType: StorageNote.self)
+        storageNote.update(with: note)
+        viewStorage.saveIfNeeded()
+
+        productReviewsRemote.whenLoadingProductReview(siteID: productReview.siteID,
+                                                      reviewID: productReview.reviewID,
+                                                      thenReturn: .success(productReview))
+        productsRemote.whenLoadingProduct(siteID: product.siteID,
+                                          productID: product.productID,
+                                          thenReturn: .success(product))
+
+        // When
+        let result = try retrieveAndWait(using: useCase, noteID: note.noteID)
+
+        // Then
+        XCTAssertTrue(result.isSuccess)
+        XCTAssertEqual(notificationsRemote.invocationCountOfLoadNotes, 0)
 
-        let useCase = makeUseCase(storage: storage)
+        let parcel = try XCTUnwrap(result.get())
+        XCTAssertEqual(parcel.note.noteID, note.noteID)
```

Comment:
nit: maybe we can compare the whole `note`? `XCTAssertEqual(parcel.note, note)` if `Note` supports `Equatable`?